### PR TITLE
Mathematical functions

### LINKED ISSUE #32: Add mathematical functions
I have the feature request from [Russian F# StackOverflow chat](http://chat.stackexchange.com/rooms/50036/f): add `arctg` and `arcctg` functions. I think that's time to review the functions we have and add those trigonometric functions we still haven't.

Please note that the ordinary notation for tangent is `tan`, but `tg` is also a common extension I think.

Probably these should go into `PredefinedTexFormulas.xml`.

### PR DESCRIPTION
**Pull request by gsomix**
- [x] Trigonometric functions (http://www.artofproblemsolving.com/wiki/index.php/LaTeX:Commands#Trigonometric_Functions)
- [x] Other functions (http://www.artofproblemsolving.com/wiki/index.php/LaTeX:Commands#Other_Functions)

op

Closes #32.

### CODE REVIEW

**Comment by gsomix** [comment]
Properly parsed groups.

groups

File: src/WpfMath/TexFormulaParser.cs [hunk 1]
```diff
@@ -550,12 +552,23 @@ private Atom AttachScripts(TexFormula formula, string value, ref int position, A
                 return atom;
 
             // Check whether to return Big Operator or Scripts.
+            var subscriptAtom = subscriptFormula == null ? null : subscriptFormula.RootAtom;
+            var superscriptAtom = superscriptFormula == null ? null : superscriptFormula.RootAtom;
```
Comment (ForNeVeR) [inline src/WpfMath/TexFormulaParser.cs]: Looks like a good place to use C# 6 syntax: `subscriptFormula?.RootAtom`, `superscriptFormula?.RootAtom`.

**Comment by ForNeVeR** [comment]
Regarding the localized function names: at the moment, I propose to add those common extensions and mark them with comment at the `PredefinedTexFormulas.xml`:

_(except for the last two, of course: they aren't functions)_

image

(Taken from the Russian [«LATEX2ε в примерах»](http://www.ccas.ru/voron/download/voron05latex.pdf) by К. В. Воронцов, p. 7.)

**Comment by ForNeVeR** [comment]
Thanks a lot!

**Comment by ForNeVeR** [comment]
Thanks for contribution! 🙌

**Pull request merged**